src/screens/login/LoginScreen.js

```diff
@@ -107,6 +108,17 @@ const LoginSteps = ({
     const magicCode = consumeRouteParameter('magicCode', route, navigation)
     const oauthCode = consumeRouteParameter('code', route, navigation)
     const onboardUrl = consumeRouteParameter('onboardUrl', route, navigation)
+    const emailVerifiedCode = consumeRouteParameter(
+      'emailVerifiedCode',
+      route,
+      navigation
+    )
+
+    if (emailVerifiedCode) {
+      // We will use it at a later stage when starting the OAuth flow
+      emailVerifiedCodeRef.current = emailVerifiedCode
```

Comment:
Here I would suggest to put this data into the `state` object for two reasons.

First because the login state is already complex so I think it is better to keep it centralised

Second because, as for the other state's entries we need to ensure it is correctly reset when login is canceled. This is done when calling `setState({step: CLOUDERY_STEP})` (i.e. in the `handleBackPress` callback or in `cancelLogin` method). With your implementation the email verified code is not cleared and will be send to the next login attempt.

Also the email code can be set only in `setInstanceData` as MagicLink is already an email based mechanism and the OIDC is not an email based one, so none of them would receive an email code.
So I think that instead of using the useRef you can safely put the code in the setState at line 190 and retrieve it in the state destructuring at line 347.

Reply:
updated